deploy/create-transfer-governor-proposal.ts

```diff
@@ -0,0 +1,358 @@
+    ]
+  );
+  targets.push(zkSyncAddress);
+  values.push(baseCostMulticall);
+  calldatas.push(l2MulticallData);
+
+  /**
+   * LAYER 2
+   */
+
+  console.log("Step 4: Renounce ownership of OwnedMulticall");
+
+  console.log("  Step 4a: Encode renounceOwnership to OwnedMulticall");
+
+  const renounceData = multicallInterface.encodeFunctionData(
+    "renounceOwnership",
+    []
+  );
+
+  console.log(
+    "  Step 4b: Estimate gas cost for renounce transaction (overestimate by 2x)"
+  );
+
+  const gasLimitRenounce = await wallet.provider.estimateL1ToL2Execute({
+    contractAddress: constants.addresses.OWNED_MULTICALL,
+    calldata: renounceData,
+    caller: utils.applyL1ToL2Alias(constants.addresses.L1_TIMELOCK),
+  });
+  const baseCostRenounce = await zkSyncContract.l2TransactionBaseCost(
+    gasPrice,
+    gasLimitRenounce,
+    utils.REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT
+  );
+
+  /**
+   * LAYER 1
+   */
+
+  console.log(
+    "  Step 4c: Encode Cross chain renounce transaction to OwnedMulticall"
+  );
+  const l2RenounceData = zkSyncContract.interface.encodeFunctionData(
+    "requestL2Transaction",
+    [
+      constants.addresses.OWNED_MULTICALL,
+      0,
+      renounceData,
+      gasLimitRenounce,
+      utils.REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT,
+      [],
+      constants.addresses.L2_GOVERNOR,
+    ]
+  );
+  targets.push(zkSyncAddress);
+  values.push(baseCostRenounce);
+  calldatas.push(l2RenounceData);
+
+  console.log("Step 5: Encode native ETH transfer to L2");
+  const gasLimitTransfer = await wallet.provider.estimateL1ToL2Execute({
+    contractAddress: constants.addresses.L2_GOVERNOR,
+    calldata: "",
+    caller: utils.applyL1ToL2Alias(constants.addresses.L1_TIMELOCK),
+    l2Value: nativeBalance,
+  });
+  const baseCostTransfer = await zkSyncContract.l2TransactionBaseCost(
+    gasPrice,
+    gasLimitTransfer,
+    utils.REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT
+  );
+  // Since previous proposal actions require sending ETH to L2 for gas,
+  // we need to calculate the remaining balance for the final transfer
+  const l2Value = //               msg.value on L2 =
+    nativeBalance - //             Starting ETH balance on L1
+    BigInt(baseCostBridge) - //    - L2 cost from Step 2 (bridge INCR)
+    BigInt(baseCostMulticall) - // - L2 cost from Step 3 (multicall)
+    BigInt(baseCostRenounce) - //  - L2 cost from Step 4 (renounce ownership)
+    BigInt(baseCostTransfer); //   - L2 cost from Step 5 (bridge ETH)
+  const l2TransferData = zkSyncContract.interface.encodeFunctionData(
+    "requestL2Transaction",
+    [
+      constants.addresses.L2_GOVERNOR,
+      l2Value,
+      "0x",
+      gasLimitTransfer,
+      utils.REQUIRED_L1_TO_L2_GAS_PER_PUBDATA_LIMIT,
+      [],
+      constants.addresses.L2_GOVERNOR,
+    ]
+  );
+  targets.push(zkSyncAddress);
+  values.push(l2Value + BigInt(baseCostTransfer)); // msg.value on L2 + base cost
+  calldatas.push(l2TransferData);
```

Comment:
Not sure I understand this entirely, is this just transferring the remaining eth in the timelock to the L2 governor?

Reply:
Yup, that's the idea. When you use the official bridge interface to transfer ETH to Era, it doesn't actually use the bridge contract at all. It just calls `requestL2Transaction` on the zkSync L1 contract, like [this tx of mine](https://etherscan.io/tx/0xde05278e88a86b43a2881e6c43954e7441ed9a8f364091b342718114f1f4b032). Note that the `msg.value` of that transaction equals the amount I input to bridge (0.1 ETH, which would be `l2Value` above) plus a small amount to cover the L2 gas cost of minting the zkSync ETH to me (`baseCostTransfer` in the code above). All of the base costs are overestimated, but the difference in each case will be sent to the `refundReceiver` specified, which should always be the new L2 governor in this script.